bundles/org.openhab.binding.lgthinq/src/main/java/org/openhab/binding/lgthinq/internal/handler/LGThinQBridgeHandler.java

```diff
@@ -0,0 +1,375 @@
+                        tokenManager.getValidRegisteredToken(bridgeName);
+                        logger.debug("Successful getting token from LG API");
+                    } catch (IOException e) {
+                        logger.debug(
+                                "I/O error accessing json token configuration file. Updating Bridge Status to OFFLINE.",
+                                e);
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                                "@text/error.toke-file-access-error");
+                        return;
+                    } catch (LGThinqException e) {
+                        logger.debug("Error accessing LG API. Updating Bridge Status to OFFLINE.", e);
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                "@text/error.lgapi-communication-error");
+                        return;
+                    }
+                }
+                if (thing.getStatus() != ThingStatus.ONLINE) {
+                    updateStatus(ThingStatus.ONLINE);
+                }
+
+                try {
+                    doConnectedRun();
+                } catch (Exception e) {
+                    logger.error("Unexpected error getting device list from LG account", e);
+                    updateStatus(ThingStatus.ONLINE, ThingStatusDetail.CONFIGURATION_PENDING,
+                            "@text/error.lgapi-getting-devices");
+                }
+
+            } finally {
+                pollingLock.unlock();
+            }
+        }
+
+        protected abstract void doConnectedRun() throws LGThinqException;
+    }
+
+    @Override
+    public void registerDiscoveryListener(LGThinqDiscoveryService listener) {
+        discoveryService = listener;
+    }
+
+    @Override
+    public void registryListenerThing(
+            LGThinQAbstractDeviceHandler<? extends CapabilityDefinition, ? extends SnapshotDefinition> thing) {
+        if (lGDeviceRegister.get(thing.getDeviceId()) == null) {
+            lGDeviceRegister.put(thing.getDeviceId(), thing);
+            // remove device from discovery list, if exists.
+            LGDevice device = lastDevicesDiscovered.get(thing.getDeviceId());
+            if (device != null && discoveryService != DUMMY_DISCOVERY_SERVICE) {
+                discoveryService.removeLgDeviceDiscovery(device);
+            }
+        }
+    }
+
+    @Override
+    public void unRegistryListenerThing(
+            LGThinQAbstractDeviceHandler<? extends CapabilityDefinition, ? extends SnapshotDefinition> thing) {
+        lGDeviceRegister.remove(thing.getDeviceId());
+    }
+
+    private final LGDevicePollingRunnable lgDevicePollingRunnable;
+
+    private LGThinQGeneralApiClientService getLgApiClient() {
+        return Objects.requireNonNull(lgApiClient, "Not expected lgApiClient null. It most likely a bug");
+    }
```

Comment:
As this method and the variable are private, there is no need for this method. You can directly access the private var.
```suggestion
```